instance/tests/models/test_openedx_theme_mixins.py

```diff
@@ -0,0 +1,130 @@
+from registration.models import BetaTestApplication
+
+
+# Tests #######################################################################
+
+class OpenEdXThemeMixinTestCase(TestCase):
+    """
+    Tests for OpenEdXThemeMixin, to check that settings from the beta application form are
+    correctly transformed into ansible variables.
+    """
+
+    @staticmethod
+    def make_test_application(instance, user):
+        """
+        Creates BetaTestApplication with test data and sample colors.
+        """
+        application = BetaTestApplication.objects.create(
+            subdomain='test',
+            instance_name='That username is mine',
+            public_contact_email='test@example.com',
+            project_description='test',
+            user=user,
+            instance=instance,
+            main_color='#001122',
+            link_color='#003344',
+            header_bg_color='#caaffe',
+            footer_bg_color='#ffff11',
+            logo='static/img/png/opencraft_logo_small.png',
+            favicon='static/img/favicon/favicon.ico',
+        )
+        return application
+
+    def test_colors_and_images_applied(self):
+        """
+        Creates a beta application with asks for some colors and logo/favicon, and checks that the generated
+        ansible variables match those colors and images.
+        """
+        # Create objects
+        OpenEdXInstanceFactory(name='Integration - test_colors_applied', deploy_simpletheme=True)
+        instance = OpenEdXInstance.objects.get()
+        user = get_user_model().objects.create_user('betatestuser', 'betatest@example.com')
+        self.make_test_application(instance, user)
+        appserver = make_test_appserver(instance)
+
+        # Test the results
+        self.assertTrue(instance.deploy_simpletheme)
+        # We check 2 times: one time just the theme vars, next whether they're in the final list
+        ansible_theme_vars = instance.get_theme_settings()
+        ansible_vars = appserver.configuration_settings
+        for variables in (ansible_theme_vars, ansible_vars):
+            parsed_vars = yaml.load(variables) or {}
+            expected_settings = {
+                'SIMPLETHEME_ENABLE_DEPLOY': True,
+                'SIMPLETHEME_SASS_OVERRIDES': [
+                    {'variable': 'link-color',
+                     'value': '#003344', },
+                    {'variable': 'header-bg',
+                     'value': '#caaffe', },
+                    {'variable': 'footer-bg',
+                     'value': '#ffff11', },
+                    {'variable': 'button-color',
+                     'value': '#001122', },
+                    {'variable': 'action-primary-bg',
+                     'value': '#001122', },
+                    {'variable': 'action-secondary-bg',
+                     'value': '#001122', },
+                ],
+                'SIMPLETHEME_STATIC_FILES_URLS': [
+                    {'url': 'static/img/png/opencraft_logo_small.png',
```

Comment:
These don't look like URLs, is that expected?

Reply:
I changed that in the other branch: https://github.com/open-craft/opencraft/blob/0eeb150c248ee65f7654c00ff1481115a73d224d/instance/tests/models/test_openedx_theme_mixins.py#L110-L114
I had to [modify](https://github.com/open-craft/opencraft/blob/0eeb150c248ee65f7654c00ff1481115a73d224d/circle.yml#L11) CircleCI to do SWIFT uploads because otherwise simple_theme can't get the logo/favicon. But the local tests in my computer don't upload to SWIFT, I think because they're using `.env.test`, and probably it should be like that (fast, mocked); there's no simple_theme in local tests and therefore noone needs to download from remote URLs. Now I allow both values: non-SWIFT (that is, non-URLs), and SWIFT (with valid URLs). I welcome ideas for cleaner approaches.